### ISSUE TITLE
Wrong indentation in Python code blocks

### ISSUE THREAD

**Issue by TerseTears** [quarto-dev/quarto-nvim#41]
Hello. Perhaps I'm missing something, but indentation does not seem to work properly inside python code blocks:

```python
for i in [1,2,3]:
  print(i)
```

Indenting instead gives:

```python
for i in [1,2,3]:
print(i)
```

**Comment by jmbuhr** [comment]
Indentation via treesitter is indeed still a bit experimental especially where the experimental indentation syntax meets the syntax for embedded languages. So I am seeing this issue as well, you are not missing something.

There have been some very recent changes to treesitter with which I have to experiment that might make this work properly (https://github.com/nvim-treesitter/nvim-treesitter/pull/4524).

**Comment by jmbuhr** [comment]
Other current issues:
```python
x = pow(2, 3)
def hello():
  print('hello')
```

Turns into
```python
x = pow(2, 3)
  def hello():
    print('hello') 
```

When the code content is highlighted follwed by `=`. When the whole code chunk is highlighted it gets even weirder:

````
```{python}
x = pow(2, 3)
  def hello():
    print('hello')
    ```
````

**Comment by georgeharker** [comment]
Can you try with #4524 as I believe it fixes these.

**Comment by jmbuhr** [comment]
Looks like this is fixed in nvim 0.9 with the latest nvim-treesitter!
Here is a version of the kickstarter configuration that uses that if you want to try it out: https://github.com/jmbuhr/quarto-nvim-kickstarter/tree/nightly
